File: VB/Q140616/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraVerticalGrid.Events
Imports DevExpress.XtraVerticalGrid
Imports System.Reflection
Imports DXSample

Namespace Q140616
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
			propertyGridControl1.SelectedObject = New CustomObject("My Custom Object", "myGOODpassword")
		End Sub

		Private Sub OnCustomRecordCellEdit(ByVal sender As Object, ByVal e As GetCustomRowCellEditEventArgs) Handles propertyGridControl1.CustomRecordCellEdit
			Dim control As PropertyGridControl = CType(sender, PropertyGridControl)
			Dim mi() As MemberInfo = control.SelectedObject.GetType().GetMember(e.Row.Properties.FieldName)
			Dim attr As PasswordPropertyTextAttribute = CType(Attribute.GetCustomAttribute(mi(0), GetType(PasswordPropertyTextAttribute)), PasswordPropertyTextAttribute)
			If attr IsNot Nothing AndAlso attr.Password Then
				e.RepositoryItem = passwordEdit
			End If
		End Sub
	End Class
End Namespace

File: CS/Q140616/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraVerticalGrid.Events;
using DevExpress.XtraVerticalGrid;
using System.Reflection;
using DXSample;

namespace Q140616 {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
            propertyGridControl1.SelectedObject = new CustomObject("My Custom Object", "myGOODpassword");
        }

        private void OnCustomRecordCellEdit(object sender, GetCustomRowCellEditEventArgs e) {
            PropertyGridControl control = (PropertyGridControl)sender;
            MemberInfo[] mi = control.SelectedObject.GetType().GetMember(e.Row.Properties.FieldName);
            PasswordPropertyTextAttribute attr = (PasswordPropertyTextAttribute)Attribute.GetCustomAttribute(mi[0], typeof(PasswordPropertyTextAttribute));
            if (attr != null && attr.Password) e.RepositoryItem = passwordEdit;
        }
    }
}